multik-cuda/src/main/kotlin/org/jetbrains/kotlinx/multik/cuda/CudaLinAlg.kt

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2020-2021 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package org.jetbrains.kotlinx.multik.cuda
+
+import jcuda.Pointer
+import jcuda.jcublas.JCublas
+import org.jetbrains.kotlinx.multik.api.LinAlg
+import org.jetbrains.kotlinx.multik.ndarray.data.*
+
+public object CudaLinAlg : LinAlg {
+    init {
+        CudaEngine
+    }
+
+    override fun <T : Number> pow(mat: MultiArray<T, D2>, n: Int): NDArray<T, D2> {
+        TODO("Not yet implemented")
+    }
+
+    override fun <T : Number> norm(mat: MultiArray<T, D2>, p: Int): Double {
+        TODO("Not yet implemented")
+    }
+
+    override fun <T : Number> inv(mat: MultiArray<T, D2>): NDArray<T, D2> {
+        TODO("Not yet implemented")
+    }
+
+    override fun <T : Number, D : Dim2> solve(a: MultiArray<T, D2>, b: MultiArray<T, D>): NDArray<T, D> {
+        TODO("Not yet implemented")
+    }
+
+    private fun <T : Number, D: Dim2> isTransposedConsistent(x: MultiArray<T, D>): Boolean {
+        return x.transpose().consistent
```

Comment:
Not the most optimal solution. You can do the same by comparing shape and strides. Make such a method in `NDArray` and call `isTranspose`.